xarray/backends/zarr.py

```diff
@@ -867,16 +868,27 @@ def store(
         """
         import zarr
 
-        existing_keys = tuple(self.zarr_group.array_keys())
+        if self._mode == "w":
```

Comment:
logic change. We now skip all these checks for mode="w". i think earlier, zarr would erase the group so all the logic below was not activated. Now that zarr does *not* erase the group, we must skip all the logic below.